crud.js

```diff
@@ -0,0 +1,69 @@
+import express, { Router } from 'express';
+import bodyParser from 'body-parser';
+import MongoQS from 'mongo-querystring';
+
+class Crud {
+  constructor(model) {
+    this.model = model;
+
+    this.router = Router();
+
+    this.queryString = new MongoQS();
+
+    this.routes = [bodyParser.json(), this.router];
+
+
+    this._setupGet();
+    this._setupPost();
+    this._setupUpdate();
+    this._setupDelete();
+  }
+
+  _setupGet() {
+    this.router.get('/', (request, response) => {
+      const query = this.queryString.parse(request.query)
+      this.model.find(query, (error, items) => { this._respond(response, error, items, 200)});
+    });
+
+    this.router.get('/:id', (request, response) => {
+      this.model.findById(request.params.id, ((error, item) => { this._respond(response, error, item, 200)}));
+    });
+  }
+
+  _setupPost() {
+    this.router.post('/', (request, response) => {
+      const item = new this.model(request.body);
```

Comment:
Probably need a secure way to C, U, and D records. Not sure if you're planning to add that eventually. Should pass along a token from the request header or something?